avahi-utils/avahi-publish.c

```diff
@@ -91,11 +91,15 @@ static void entry_group_callback(AvahiEntryGroup *g, AvahiEntryGroupState state,
                 n = avahi_alternative_host_name(config->name);
             }
 
-            fprintf(stderr, _("Name collision, picking new name '%s'.\n"), n);
-            avahi_free(config->name);
-            config->name = n;
-
-            register_stuff(config);
+	    if (n) {
+		fprintf(stderr, _("Name collision, picking new name '%s'.\n"), n);
+		avahi_free(config->name);
+		config->name = n;
+
+		register_stuff(config);
+	    } else
+		fprintf(stderr, _("Name collision, failed to pick a new name for '%s'.\n"),
+				config->name);
```

Comment:
@pemensik could you move the last commit to a separate PR? It changes a few different things in a few different places and I think it would be better to review them separately (one by one probably). For example this change prevents avahi-publish from crashing (https://github.com/avahi/avahi/issues/478) but it doesn't fix the issue in the sense that `avahi-publish` still fails to pick the new name there.

I'm not sure what the test does either. It doesn't trigger the overflow. Could you drop it too? OSS-Fuzz should test it on x86_64 and i386 and I can add a test calling busctl to the smoke-test calling busctl later.

The first commit looks good (even though I think ideally  but I wonder if it should count from 1 instead of 2? I'm not sure where 2 came from. Could you also add "fixes https://github.com/avahi/avahi/issues/491" to the commit message?

Reply:
I extracted the first commit, replaced tabs with spaces there and added a smoke test on top: https://github.com/avahi/avahi/pull/568. The other commits can be reviewed here later.